src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/FileSystemEgressExtensionTests.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using Microsoft.Diagnostics.Tools.Monitor;
+using Microsoft.Diagnostics.Tools.Monitor.Egress;
+using Microsoft.Diagnostics.Tools.Monitor.Egress.Configuration;
+using Microsoft.Diagnostics.Tools.Monitor.Egress.FileSystem;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Moq;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.UnitTests
+{
+    public sealed class FileSystemEgressExtensionTests
+    {
+        const string ProviderName = "TestProvider";
+
+        private static readonly string CopyBufferSize_RangeErrorMessage = CreateRangeMessage<int>(
+            nameof(FileSystemEgressProviderOptions.CopyBufferSize),
+            FileSystemEgressProviderOptions.CopyBufferSize_MinValue.ToString(),
+            FileSystemEgressProviderOptions.CopyBufferSize_MaxValue.ToString());
+
+        private readonly ITestOutputHelper _outputHelper;
+
+        public FileSystemEgressExtensionTests(ITestOutputHelper outputHelper)
+        {
+            _outputHelper = outputHelper;
+        }
+
+        [Fact]
+        public async Task FileSystemEgressExtension_EmptyConfiguration_ThrowsException()
+        {
+            // Arrange
+            Mock<IEgressConfigurationProvider> mockConfigurationProvider = new();
+            mockConfigurationProvider.Setup(provider => provider.GetProviderConfigurationSection(EgressProviderTypes.FileSystem, ProviderName))
+                .Returns(CreateConfigurationSection());
+
+            Mock<ILogger<FileSystemEgressExtension>> mockLogger = new();
+
+            FileSystemEgressExtension extension = new(mockConfigurationProvider.Object, mockLogger.Object);
+
+            // Act & Assert
+            OptionsValidationException exception = await Assert.ThrowsAsync<OptionsValidationException>(
+                () => extension.EgressArtifact(ProviderName, null, (stream, token) => Task.CompletedTask, CancellationToken.None));
+            string errorMessage = Assert.Single(exception.Failures);
+            Assert.Equal(CreateRequiredMessage(nameof(FileSystemEgressProviderOptions.DirectoryPath)), errorMessage);
+        }
+
+        [Fact]
+        public async Task FileSystemEgressExtension_CopyBufferSizeZero_ThrowsException()
```

Comment:
nit: Is it worth having a test where the `CopyBufferSize` **is** valid, like for `DirectoryPath`?